pycbc/inference/likelihood.py

```diff
@@ -0,0 +1,192 @@
+                sorted(self._data.keys()):
+            raise ValueError("waveform generator's detectors (%s) " %(
+                ','.join(sorted(waveform_generator.detector_names))) +
+                "does not match data (%s)" %(
+                ','.join(sorted(self._data.keys()))))
+        # check that the data sets all have the same lengths
+        dlens = numpy.array([len(d) for d in data.values()])
+        if not all(dlens == dlens[0]):
+            raise ValueError("all data must be of the same length")
+        N = dlens[0]
+        # we'll use the first data set for setting values
+        d = data.values()[0]
+        # figure out the kmin, kmax to use
+        kmin, kmax = filter.get_cutoff_indices(f_lower, f_upper, d.delta_f,
+            (N-1)*2)
+        self._kmin = kmin
+        self._kmax = kmax
+        if norm is None:
+            norm = 4*d.delta_f
+        # we'll store the weight to apply to the inner product
+        if psd is None:
+            self._weight = norm*Array(numpy.ones(N))
+        else:
+            # temporarily suppress numpy divide by 0 warning
+            numpy.seterr(divide='ignore')
+            self._weight = norm/psd
+            numpy.seterr(divide='warn')
+        # compute <d, d>
+        self._dd = {det:
+            d[kmin:kmax].inner(d[kmin:kmax]*self._weight[kmin:kmax]).real/2.
+            for det,d in self._data.items()}
+
+
+    @property
+    def waveform_generator(self):
+        """Returns the waveform generator that was set."""
+        return self._waveform_generator
+
+    @property
+    def data(self):
+        """Returns the data that was set."""
+        return self._data
+
+    def loglikelihood(self, params):
+        """This function should return the log likelihood of the given params.
+        """
+        raise ValueError("Likelihood function not set.")
+
+
+class GaussianLikelihood(_BaseLikelihoodEvaluator):
+    r"""Computes the log likelihood for the given parameters using:
+
+    .. math:: \log \mathcal{L} \propto -\frac{1}{2}\sum_{i} \left<\mathbf{h}_i(\mathbf{\theta}) - \mathbf{s}_i | \mathbf{h}_i(\mathbf{\theta}) - \mathbf{s}_i \right>
+
+    where :math:`\mathbf{h}_i(\mathbf{\theta})` and :math:`\mathbf{s}_i`
+    are the model waveform vector with parameters :math:`\mathbf{\theta}`
+    and the data, respectively, in the :math:`i`th detector (the sum is
+    over the detectors), and the inner product is given by:
+
+    .. math:: \left<a | b\right> = 4\Re \int \tilde{a}(f) \tilde{b}(f) \mathrm{d}f.
+
+    No analytic marginalization is done, and the data is assumed to not be
+    whitened.
+
+    For details on initialization parameters, see _BaseLikelihoodEvaluator.
+
+    Examples
+    --------
+    Create a signal, and compute the log likelihood for a template with the same
+    parameters at the same time (the log likelihood should be zero in this case):
+    >>> from pycbc import psd as pypsd, inference, waveform
```

Comment:
This is python 2.7 syntax. This won't work with python 2.6.6 which is installed on the clusters, eg. sugar. Can we change this to: ``from pycbc import psd as pypsd; from pycbc import inference, waveform`` for python 2.6.6?